src/autoproj.ts

```diff
@@ -137,6 +143,28 @@ export class Workspace
             })
         })
     }
+
+    async which(cmd: string)
+    {
+        let options: child_process.SpawnOptions = {};
+        options.env = { AUTOPROJ_CURRENT_ROOT: this.root };
+        let subprocess = child_process.spawn(this.autoprojExePath(), ['which', cmd], options);
+        let path = '';
+        subprocess.stdout.on('data', (buffer) => {
+            path = path.concat(buffer.toString());
+        })
+
+        return new Promise<string>((resolve, reject) => {
+            subprocess.on('close', (code, signal) => {
+                if (code !== 0) {
+                    reject(`cannot find ${cmd} in the workspace`)
```

Comment:
Will update, could you explain why ?

Reply:
Multiple reasons:

1. Consistency (we are already doing it like that in our code base and it is also how vscode rejects its promises);
2. Backtrace;
3. Consider the scenario:

```ts
new Promise<void>((resolve, reject) => {
    doSomething(); // this may throw
    if (!expectSomething())
    {
        reject("failure");
    }
});
```

To properly handle the rejected promise the caller has to expect both a string and an `Error` as the reason because throwing an exception in the promise automatically rejects it with the exception as the reason...

Reply:
(pushed a fixed version, which actually unit tests which now :P)